src/main/java/org/jenkinsci/plugins/durabletask/FileMonitoringTask.java

```diff
@@ -179,9 +179,14 @@ String getCharset() {
         protected FileMonitoringController(FilePath ws) throws IOException, InterruptedException {
             // can't keep ws reference because Controller is expected to be serializable
             ws.mkdirs();
-            FilePath cd = tempDir(ws).child("durable-" + Util.getDigestOf(UUID.randomUUID().toString()).substring(0,8));
-            cd.mkdirs();
-            controlDir = cd.getRemote();
+            FilePath tmpDir = WorkspaceList.tempDir(ws);
```

Comment:
Note that this will only work on top of one of the core PRs under discussion. Until then, the `tempDir` call with fail with an NPE. As far as I know, no one has complained about this so far, but if you want to handle _current_ core versions you could

```suggestion
            FilePath tmpDir = /* TODO pending JENKINS-61197 fix in baseline */ ws.getParent() != null ? WorkspaceList.tempDir(ws) : null;
```